core/src/main/java/org/eclipse/hono/adapter/AdapterConfig.java

```diff
@@ -27,6 +30,10 @@
 
     private final Vertx vertx = Vertx.vertx();
 
+    @Autowired(required=false)
+    @Qualifier("registration")
+    private ClientConfigProperties registrationServiceClientConfig;
```

Comment:
An alternative to using the two `@Autowired(required=false)` fields would be to also use the `@ConditionalOnProperty` annotation on the two methods where optional parameters are used.

That would mean having the line
`@ConditionalOnProperty(prefix = "hono.registration", name = "host")`
occurring in 4 places. 
While I would prefer not having that code duplication, it would remove the null-checks of the `@Autowired(required=false)` solution. So I'm fine with either of these solutions.
What do you think?

(Another thing I thought about was using `ConditionalOnBean`. But that is dependent on the load order of beans and only recommended in `@AutoConfiguration` classes.)